File: mksite.Server/Models/ExperienceModel.cs
namespace mksite.Server.Models;

public class ExperienceListModel
{
    public int Id { get; set; }
    public string Organization { get; set; }
    public string Role { get; set; }
    public string Duration { get; set; }
    public string GeneralDescription { get; set; }
    public string[] ItemizedDescription { get; set; }
    public string? Link { get; set; }
}

public class ExperienceModel { 
    public ExperienceListModel[] ExperienceList { get; set; }
    public string Title { get; set; }
    public string Introduction { get; set; }
    public CompetencyModel[]? CoreCompetencies { get; set; }

 }

 public class CompetencyModel {
    public int Id { get; set; }
    public string Title { get; set; }
    public string Url { get; set; } 
    public string? Details { get; set; } 
    public string Category { get; set; }
 }

File: mksite.client/src/models/ExperienceModel.ts
export type ExperienceListModel = {
    id: number,
    organization: string,
    role: string, 
    duration: string,
    generalDescription: string,
    itemizedDescription: string[],
    link?: string
}

export type ExperienceModel = {
    title: string;
    introduction: string;
    coreCompetencies?: CompetencyModel[];
    experienceList: ExperienceListModel[];
    certifications?: CertificateModel[];
}

export type CompetencyModel = {
    id: number;
    title: string;
    url: string;
    details?: string;
    category: string;
}

export type CertificateModel = {
    id: number;
    courseName: string;
    institution: string;
    details: string;
    link?: string;
    github?: string;
}